firebase-firestore/src/androidTest/java/com/google/firebase/firestore/TransactionTest.java

```diff
@@ -637,6 +705,29 @@ public void testDoesNotRetryOnPermanentError() {
     assertEquals(1, count.get());
   }
 
+  @Test
+  public void testRetryOnAlreadyExistsError() {
+    final FirebaseFirestore firestore = testFirestore();
+    DocumentReference doc = firestore.collection("foo").document();
+    AtomicInteger retryCount = new AtomicInteger(0);
+    waitFor(
+        firestore.runTransaction(
+            transaction -> {
+              retryCount.getAndIncrement();
+              transaction.get(doc);
+              // Do a write outside of the transaction.
+              if (retryCount.get() == 1) waitFor(doc.set(map("count", retryCount.get())));
```

Comment:
Please use a "dummy" key/value pair here instead of `map("count", retryCount.get())`, which seems to imply that the value of `retryCount` is relevant.

e.g. `map("foo1", "bar1")`

Then maybe use `map("foo2", "bar2")` to `transaction.set()` also. What I'm looking for is a way for these values to clearly suggest that the key/value pair itself is completely irrelevant, and that the values set by `doc.set()` is completely independent from those set by `transaction.set()`.